lib-injection/sitecustomize.py

```diff
@@ -51,13 +173,68 @@ def _inject():
         _log("ddtrace_pkgs path is %r" % pkgs_path, level="debug")
         _log("ddtrace_pkgs contents: %r" % os.listdir(pkgs_path), level="debug")
 
-        python_version = ".".join(str(i) for i in sys.version_info[:2])
-        if python_version not in installable_py_versions:
-            _log(
-                f"This version of ddtrace does not support single step instrumentation with python {python_version} "
-                f"(supported versions: {installable_py_versions}), aborting",
-                level="error",
+        # check installed packages against allow list
+        incompatible_packages = {}
+        for package_name, package_version in installed_packages.items():
+            if not package_is_compatible(pkgs_allow_list, package_name, package_version):
+                incompatible_packages[package_name] = package_version
+
+        if incompatible_packages:
+            _log("Found incompatible packages: %s." % incompatible_packages, level="debug")
+            integration_incomp = True
+            if not force_inject:
+                _log("Aborting dd-trace-py instrumentation.", level="debug")
+
+                for key, value in incompatible_packages.items():
+                    telemetry_data.append(
+                        create_count_metric(
+                            "library_entrypoint.abort.integration",
+                            [
+                                "integration_name:" + key,
+                                "integration_version:" + value,
+                                "min_supported_version:" + str(pkgs_allow_list[key]),
+                            ],
+                        )
+                    )
+
+            else:
+                _log(
+                    "DD_INJECT_FORCE set to True, allowing unsupported integrations.",
+                    level="debug",
+                )
+        if not runtime_version_is_supported(runtimes_allow_list, python_runtime, python_version):
+            _log("Found incompatible runtime: %s %s." % (python_runtime, python_version), level="debug")
```

Comment:
Can we add supported_runtimes?
“Found incompatible runtime <runtime_name>, <runtime_version>. Supported runtimes: <supported_runtimes>”